test-d/exact.ts

```diff
@@ -537,3 +537,19 @@ import type {Exact, Opaque} from '../index';
 	// @ts-expect-error
 	function_(withExcessSurname);
 }
+
+// Spec - recursive type with union
+// @see https://github.com/sindresorhus/type-fest/issues/948
+{
+	type A = {
+		a: A;
```

Comment:
Sorry, I don't understand this type. Looks like this is an infinite recursive type.

Do you actually mean the following?

```
type A = {
  a: Expected;
}
```

Reply:
Ah, yes I think that's what I meant

Reply:
Fixed. I think it would also test the same case with `a?: A`. As long as there's _some_ sort of recursion in either of the types. TS doesn't even really care about the infinite recursion unless you try to actually instantiate or resolve the type 🤷